### DISCUSSION TITLE
Integration with https://labels.electrum.org/

### DISCUSSION THREAD

**Discussion by ser** [btcpayserver/btcpayserver#662] [Plugins & integrations]
It would be fantastic to have information about invoices directly in your wallet. 

https://labels.electrum.org/ is a free service and we can use it as well.

The code is pretty straightforward, it utilises xpub to encrypt data:
https://github.com/spesmilo/electrum/blob/master/electrum/plugins/labels/labels.py#L46

**Comment by Kukks** [comment]
The site is down?

**Comment by ser** [comment]
It is not down, it just needs a client to use the API.

**Comment by Kukks** [comment]
are there any proper docs or anything that describes what it does? I'm not a python guy and don't care for it.

**Comment by ser** [comment]
I have the same situation, if I were a C# guy I would just send a PR :) :)

The server is written in golang:
https://github.com/maran/electrum-sync-server

But there is no written specification I am afraid - I can write one if someone would need it, though.

In short: the server accepts anonymous read/writes of short text fields to keys encoded from xpub, so you can keep memos related to tx/addresses on the server side. Thus when you use any connected electrum wallet, you have a nice description of your transactions.

**Comment by ser** [comment]
BTW: I use it in my poor-man version of btcpayserver: 
https://github.com/spesmilo/electrum-merchant

**Comment by astupidmoose** [comment]
@ser Is the label server compatible with any version of electrum (BTC, LTC, Dash)? 

As someone who uses electrum, I actually think this would be super useful and cool. I'm going to play around with the server aspect this weekend if I have time.

**Comment by ser** [comment]
Yes, I am using it in all mentioned versions.

**Comment by pavlenex** [comment]
Ok, closing this one since we have labels integrated.

**Comment by ser** [comment]
As I can't see a link to a code, is it an electrum label or home grown solution?

**Comment by astupidmoose** [comment]
> As I can't see a link to a code, is it an electrum label or home grown solution?

Labels were implemented into BTCpay's internal wallets. That's not really what this issue was about, but I assume it's being closed due to a homegrown solution plus lack of activity on this issue.

**Comment by ser** [comment]
OK, it is unfortunately completely not what is that issue about, as it does not give overview of transaction in Electrum wallet. Using btcpayserver to browse transactions is annoying if you could have them in a wallet.

**Comment by pavlenex** [comment]
I can re-open the issue if there's still need for it, and someone actually wants to tackle it and code it.

Also we try to discourage the use of Electrum, since plenty of noobs use their public servers and have no idea how to use it privately.

**Comment by ser** [comment]
I have not seen any case of public Electrum servers stealing money. There was a security bug in Electrum which was related to handling the servers, but it was clearly a bug which could happen in any software. 

In my opinion leaving it open is better than closing, as it is a feature request, not a bug.
